lib/Discovery_Schema.cpp

```
@@ -0,0 +1,2243 @@
+}
+
+Discovery_Schema::Discovery_Schema(const std::string& path)
+	: db(NULL), db_path(path)
+{
+}
+
+Discovery_Schema::~Discovery_Schema() {
+	close();
+}
+
+int Discovery_Schema::init() {
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)